From 4b0c9025388d3f448f1658b4f4ff647ce1f93507 Mon Sep 17 00:00:00 2001
From: Yotam Ben Dosa <ybendosa@habana.ai>
Date: Tue, 15 Nov 2022 14:06:59 +0200
Subject: [PATCH] [SW-110272] gaudi2: add eROM rec efuse cfg FIT

Add a target for creating a FIT that enables eROM rec in EFUSE.

Change-Id: Ica9773a4e9ed116e007dde5252ecbe99ff3bffb2
Signed-off-by: Yotam Ben Dosa <ybendosa@habana.ai>
---

diff --git a/scripts/make/gaudi2/Makefile b/scripts/make/gaudi2/Makefile
index 3cc76fc..4633536 100644
--- a/scripts/make/gaudi2/Makefile
+++ b/scripts/make/gaudi2/Makefile
@@ -154,14 +154,16 @@
 	${HL_DEV}-otp.itb				\
 	${HL_DEV}-otp-local.itb		\
 	${HL_DEV}-eeprom-recovery.itb	\
-	${HL_DEV}-hl225b_enable.itb
+	${HL_DEV}-hl225b_enable.itb	\
+	${HL_DEV}-erom_recovery_enable.itb
 
 .PHONY: gaudi2-internal-fit-create
 gaudi2-internal-fit-create: gaudi2-nic-en-create	\
 				gaudi2-eeprom-create-default gaudi2-write-tpm-devid-create	\
 				gaudi2-slt-set-create gaudi2-security-enable-create	\
 				gaudi2-otp-create gaudi2-otp-local-create	\
-				gaudi2-eeprom-create-recovery gaudi2-hl225b-enable-create
+				gaudi2-eeprom-create-recovery gaudi2-hl225b-enable-create	\
+				gaudi2-erom-recovery-enable-create
 
 	@$(call hl-check-files,$(GAUDI2_INTERNAL_FITS))
 	@tar czf $(IMAGES_DIR)/${HL_DEV}-int-fit.tgz -C $(IMAGES_DIR)/ $(GAUDI2_INTERNAL_FITS)
diff --git a/scripts/make/hl-common/Makefile b/scripts/make/hl-common/Makefile
index 5e87cf3..3762362 100644
--- a/scripts/make/hl-common/Makefile
+++ b/scripts/make/hl-common/Makefile
@@ -168,6 +168,12 @@
 	@$(if $(filter $(call hl-device, $@), $(HL_SEC_FIT_CREATE_ARCH)), \
 		${BR2_EXTERNAL}/board/habanalabs/common/create-simple.sh erom_tpm_enable fw-loader-conf no-use-elbi no-use-fwupdate load-bootloader)
 
+.PHONY: %-erom-recovery-enable-create
+%-erom-recovery-enable-create:
+	@$(if $(filter $(call hl-device, $@), $(HL_SEC_FIT_CREATE_ARCH)), \
+		${BR2_EXTERNAL}/board/habanalabs/common/create-simple.sh erom_recovery_enable fw-loader-conf no-use-elbi no-use-fwupdate load-bootloader)
+
+
 .PHONY: %-security-enable-install
 %-security-enable-install:
 	@$(if $(filter $(call hl-device, $@), $(HL_SEC_FIT_CREATE_ARCH)), \
